### PR TITLE
attempt very awful trick for performance boost

### PR DESCRIPTION
**Pull request by andrewthad**

### CODE REVIEW

**Comment by chessai** [comment]
How big is the boost

**Comment by andrewthad** [comment]
[The boost](https://github.com/goolord/url-bytes/pull/1) is for `url-bytes`. It makes the URL parser about 20% faster and eliminates all allocations other than the result.

File: src/Data/Bytes/Parser/Rebindable.hs [hunk 1]
```diff
@@ -0,0 +1,252 @@
+{-# language FlexibleInstances #-}
+{-# language MagicHash #-}
+{-# language MultiParamTypeClasses #-}
+{-# language RankNTypes #-}
+{-# language ScopedTypeVariables #-}
+{-# language TypeInType #-}
+{-# language UnboxedSums #-}
+{-# language UnboxedTuples #-}
+
+-- | This is a giant hack to let authors of high-performance use
```
Comment (chessai) [inline src/Data/Bytes/Parser/Rebindable.hs]: i think you meant to insert something like 'code' after 'high-performance'

Comment (andrewthad) [reply src/Data/Bytes/Parser/Rebindable.hs]: I've rewritten this to make it more clear what this is for.

**Comment by andrewthad** [comment]
If there are no other concerns, I'll merge this tomorrow.

**Pull request merged**